go/tasks/pluginmachinery/core/exec_metadata.go

```diff
@@ -30,6 +30,7 @@ type TaskExecutionMetadata interface {
 	GetOwnerReference() v12.OwnerReference
 	GetOverrides() TaskOverrides
 	GetLabels() map[string]string
+	GetMaxAttempt() uint32
```

Comment:
@kumare3 Are you fine with the naming? Or prefer specific - GetNodeMaxAttempts() or something.